src/run/next.cts

```diff
@@ -101,7 +105,7 @@ export async function getMockedRequestHandler(...args: Parameters<typeof getRequ
         // only try to get .html files from the blob store
         if (typeof path === 'string' && path.endsWith('.html')) {
           const cacheStore = getMemoizedKeyValueStoreBackedByRegionalBlobStore()
-          const relPath = relative(resolve('.next/server/pages'), path)
+          const relPath = relative(resolve(nextConfig.distDir, 'server/pages'), path)
```

Comment:
not related to Node.js Middleware support, but this fixes serving static files if non-default `distDir` is used (test setup for Node.js middleware variants is hitting those conditions)